libsolidity/interface/CompilerStack.cpp

```diff
@@ -1219,9 +1219,25 @@ void CompilerStack::resolveImports()
 		sourceOrder.push_back(_source);
 	};
 
+	vector<PragmaDirective const*> experimentalPragmaDirectives;
 	for (auto const& sourcePair: m_sources)
+	{
 		if (isRequestedSource(sourcePair.first))
 			toposort(&sourcePair.second);
+		if (sourcePair.second.ast && sourcePair.second.ast->experimentalParsingEnabled())
+			for (ASTPointer<ASTNode> const& node: sourcePair.second.ast->nodes())
+				if (PragmaDirective const* pragma = dynamic_cast<PragmaDirective*>(node.get()))
+					if (pragma->literals().size() >=2 && pragma->literals()[0] == "experimental" && pragma->literals()[1] == "solidity")
+						experimentalPragmaDirectives.push_back(pragma);
+	}
+
+	if (experimentalPragmaDirectives.size() != m_sources.size())
+		for (auto&& pragma: experimentalPragmaDirectives)
+			m_errorReporter.parserError(
+				2141_error,
+				pragma->location(),
+				"File declares \"pragma solidity solidity\". If you want to enable the experimental mode, all source units must include the pragma."
```

Comment:
Shouldn't this be the following?
```suggestion
				"File declares \"pragma experimental solidity\". If you want to enable the experimental mode, all source units must include the pragma."
```

Reply:
ops, just saw that @ekpyron already mentioned that.